### DISCUSSION TITLE
Unable upload devices by using csv file

### DISCUSSION THREAD

**Discussion by aaanserali** [netbox-community/netbox#14257] [Help Wanted!]
I am unable to upload devices using the CSV file and am encountering the following error

![image](https://github.com/netbox-community/netbox/assets/44871704/4b02118a-3546-4152-9bdc-c4d31c83ecb7)

**Comment by candlerb** [comment]
The error seems pretty clear to me: your file is not valid UTF-8.  It even tells you exactly where the error is, as a byte offset within the file.

Hex byte 0xA0 is a non-breaking space in some encodings, e.g. [Windows-1252](https://en.wikipedia.org/wiki/Windows-1252), but if you want to feed this to Netbox you would have to use an external utility to transcode it to UTF-8 first.

**Comment by mtinberg** [comment]
Just to show how encoding can work and how you can use iconv to convert text from one encoding to another.

$ python3
Python 3.6.8 (default, Aug 10 2023, 17:01:17)
[GCC 8.5.0 20210514 (Red Hat 8.5.0-20)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print(f"asdf{chr(0xa0)}bar")
asdf bar

save that to a text file

$ file testencoding.txt
testencoding.txt: ISO-8859 text, with CRLF line terminators

$ cat testencoding.txt
asdf�bar

$ iconv  --from-code=ISO-8859-1 --to-code=UTF-8 testencoding.txt  > testencoding2.txt

$ file testencoding*
testencoding2.txt: UTF-8 Unicode text, with CRLF line terminators
testencoding.txt:  ISO-8859 text, with CRLF line terminators

$ hexdump  --canonical testencoding.txt
00000000  61 73 64 66 a0 62 61 72  0d 0a                    |asdf.bar..|
0000000a
$ hexdump  --canonical testencoding2.txt
00000000  61 73 64 66 c2 a0 62 61  72 0d 0a                 |asdf..bar..|
0000000b

$ more testencoding.txt testencoding2.txt
::::::::::::::
testencoding.txt
::::::::::::::
asdf�bar
::::::::::::::
testencoding2.txt
::::::::::::::
asdf bar





—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
